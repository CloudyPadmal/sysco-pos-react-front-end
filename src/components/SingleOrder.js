import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import SingleItem from './SingleItem';

/**
 * Container for the list of todos
 */
class SingleOrder extends Component {

    viewThisOrder = () => {
        this.props.viewThisOrder(this.props._id);
        this.props.history.push('/view_order');
    }

    editThisOrder = () => {
        this.props.editThisOrder(this.props._id);
        this.props.history.push('/edit_order');
    }

    render() {
        let { _id, items } = this.props.singleOrder;
        let totalCost = 0;
        for (var i = 0; i < items.length; i++) {
            let { quantity, price } = items[i];
            totalCost += (quantity * price);
        }
        return (
            <div className="card border-dark shadow" style={{ margin: '10px 0px' }}>
                <div className="card-header text-white bg-dark" style={{ padding: '.75em .1em' }}>
                    <div className="row" style={{ width: '100%', margin: '0px' }}>
                        <div className="col-10 d-flex justify-content-start">
                            Order ID: {_id}</div>
                        <div className="col-2 d-flex justify-content-end">
                            <b>Rs. {totalCost.toFixed(2)}</b>
                        </div>
                    </div>
                </div>
                <ul className="list-group list-group-flush">
                    {items.map((item) => (
                        <SingleItem key={item._id} singleItem={item} />
                    ))}
                </ul>
                <div className="card-footer d-flex justify-content-end">
                    <Link to="#" className="card-link" onClick={this.viewThisOrder}>View</Link>
                    <Link to="#" className="card-link edit-order" onClick={this.editThisOrder}>Edit</Link>
                    <Link to="#" className="card-link delete" onClick={this.props.deleteThisOrder}>Delete</Link>
                </div>
            </div>
        );
    }
}

export default withRouter(SingleOrder);