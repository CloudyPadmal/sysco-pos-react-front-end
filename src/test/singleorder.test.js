import React from 'react';
import { mountWrap } from './enzyme';

import SingleOrder from '../components/SingleOrder';

describe('Single Order', () => {

    it('displays a single order', () => {
        const wrapper = mountWrap(<SingleOrder
            key={213123123}
            ORDER={{ _id: 4324, items: [] }}
            PREPARE_TO_EDIT_OR_VIEW_THIS_ORDER={() => { }}
            DELETE_THIS_ORDER={() => { }}
            SET_THIS_ORDER_AS_CURRENT={() => { }} />);
        expect(wrapper).toMatchSnapshot();
    });
});