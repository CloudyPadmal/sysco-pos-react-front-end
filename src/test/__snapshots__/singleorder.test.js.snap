// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Order displays a single order 1`] = `
<withRouter(SingleOrder)
  DELETE_THIS_ORDER={[Function]}
  ORDER={
    Object {
      "_id": 4324,
      "items": Array [],
    }
  }
  PREPARE_TO_EDIT_OR_VIEW_THIS_ORDER={[Function]}
  SET_THIS_ORDER_AS_CURRENT={[Function]}
>
  <Route>
    <SingleOrder
      DELETE_THIS_ORDER={[Function]}
      ORDER={
        Object {
          "_id": 4324,
          "items": Array [],
        }
      }
      PREPARE_TO_EDIT_OR_VIEW_THIS_ORDER={[Function]}
      SET_THIS_ORDER_AS_CURRENT={[Function]}
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={Object {}}
      match={Object {}}
    >
      <div
        className="card border-dark shadow"
        style={
          Object {
            "margin": "10px 0px",
          }
        }
      >
        <div
          className="card-header text-white bg-dark"
          style={
            Object {
              "padding": ".75em .1em",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "margin": "0px",
                "width": "100%",
              }
            }
          >
            <div
              className="col-10 d-flex justify-content-start"
              style={
                Object {
                  "textAlign": "initial",
                }
              }
            >
              Order ID: 
              4324
            </div>
            <div
              className="col-2 d-flex justify-content-end"
              style={
                Object {
                  "textAlign": "initial",
                }
              }
            >
              <b>
                Rs. 
                0.00
              </b>
            </div>
          </div>
        </div>
        <ul
          className="list-group list-group-flush"
        />
        <div
          className="card-footer d-flex justify-content-end"
        >
          <Link
            className="card-link"
            onClick={[Function]}
            replace={false}
            to="#"
          >
            <a
              className="card-link"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fas fa-list-alt"
              />
               View
            </a>
          </Link>
          <Link
            className="card-link edit-order"
            onClick={[Function]}
            replace={false}
            to="#"
          >
            <a
              className="card-link edit-order"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fas fa-pen"
              />
               Edit
            </a>
          </Link>
          <Link
            className="card-link delete"
            onClick={[Function]}
            replace={false}
            to="#"
          >
            <a
              className="card-link delete"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fas fa-trash"
              />
               Delete
            </a>
          </Link>
        </div>
      </div>
    </SingleOrder>
  </Route>
</withRouter(SingleOrder)>
`;
